feature/interface/aggregate/ate_tests/aggregate_test/aggregate_test.go

```diff
@@ -0,0 +1,441 @@
+//     subnet 192.0.2.4/30 2001:db8::4/126
+//
+// Note that the first (.0, .4) and last (.3, .7) IPv4 addresses are
+// reserved from the subnet for broadcast, so a /30 leaves exactly 2
+// usable addresses.  This does not apply to IPv6 which allows /127
+// for point to point links, but we use /126 so the numbering is
+// consistent with IPv4.
+//
+// A traffic flow is configured from ate:port1 as source and ate:port{2-9}
+// as destination.
+const (
+	plen4 = 30
+	plen6 = 126
+)
+
+var (
+	dutSrc = attrs.Attributes{
+		Desc:    "dutsrc",
+		IPv4:    "192.0.2.1",
+		IPv6:    "2001:db8::1",
+		IPv4Len: plen4,
+		IPv6Len: plen6,
+	}
+
+	ateSrc = attrs.Attributes{
+		Name:    "atesrc",
+		IPv4:    "192.0.2.2",
+		IPv6:    "2001:db8::2",
+		IPv4Len: plen4,
+		IPv6Len: plen6,
+	}
+
+	dutDst = attrs.Attributes{
+		Desc:    "dutdst",
+		IPv4:    "192.0.2.5",
+		IPv6:    "2001:db8::5",
+		IPv4Len: plen4,
+		IPv6Len: plen6,
+	}
+
+	ateDst = attrs.Attributes{
+		Name:    "atedst",
+		IPv4:    "192.0.2.6",
+		IPv6:    "2001:db8::6",
+		IPv4Len: plen4,
+		IPv6Len: plen6,
+	}
+)
+
+const (
+	lagTypeLACP   = telemetry.IfAggregate_AggregationType_LACP
+	lagTypeSTATIC = telemetry.IfAggregate_AggregationType_STATIC
+)
+
+type testCase struct {
+	minlinks uint16
+	lagType  telemetry.E_IfAggregate_AggregationType
+
+	dut *ondatra.DUTDevice
+	ate *ondatra.ATEDevice
+	top *ondatra.ATETopology
+
+	dutPorts []*ondatra.Port
+	atePorts []*ondatra.Port
+	aggID    string
+	l3header []ondatra.Header
+}
+
+func (*testCase) configSrcDUT(i *telemetry.Interface, a *attrs.Attributes) {
+	i.Description = ygot.String(a.Desc)
+	if *deviations.InterfaceEnabled {
+		i.Enabled = ygot.Bool(true)
+	}
+
+	s := i.GetOrCreateSubinterface(0)
+	s4 := s.GetOrCreateIpv4()
+	if *deviations.InterfaceEnabled {
+		s4.Enabled = ygot.Bool(true)
+	}
+	a4 := s4.GetOrCreateAddress(a.IPv4)
+	a4.PrefixLength = ygot.Uint8(plen4)
+
+	s6 := s.GetOrCreateIpv6()
+	if *deviations.InterfaceEnabled {
+		s6.Enabled = ygot.Bool(true)
+	}
+	s6.GetOrCreateAddress(a.IPv6).PrefixLength = ygot.Uint8(plen6)
+}
+
+func (tc *testCase) configDstAggregateDUT(i *telemetry.Interface, a *attrs.Attributes) {
+	tc.configSrcDUT(i, a)
+	i.Type = ieee8023adLag
+	g := i.GetOrCreateAggregation()
+	g.LagType = tc.lagType
+	g.MinLinks = ygot.Uint16(tc.minlinks)
+}
+
+var portSpeed = map[ondatra.Speed]telemetry.E_IfEthernet_ETHERNET_SPEED{
+	ondatra.Speed10Gb:  telemetry.IfEthernet_ETHERNET_SPEED_SPEED_10GB,
+	ondatra.Speed100Gb: telemetry.IfEthernet_ETHERNET_SPEED_SPEED_100GB,
+	ondatra.Speed400Gb: telemetry.IfEthernet_ETHERNET_SPEED_SPEED_400GB,
+}
+
+func (tc *testCase) configDstMemberDUT(i *telemetry.Interface, p *ondatra.Port) {
+	i.Description = ygot.String(p.String())
+	i.Type = ethernetCsmacd
+
+	if *deviations.InterfaceEnabled {
+		i.Enabled = ygot.Bool(true)
+	}
+
+	e := i.GetOrCreateEthernet()
+	e.AggregateId = ygot.String(tc.aggID)
+	// Speed of 99 Gbps is a temporary workaround indicating that the
```

Comment:
This might be a little tricky. Ondatra has a pending change that adds PMD() to ondatra.Port. It's staged internally but not yet open sourced.

https://pkg.go.dev/github.com/openconfig/ondatra#Port

When it's published, the code should look like this:

```
if p.PMD() == "PMD_100G_FR" {
  ...
}
```

What do you think, leave the speed 99 workaround for now and follow up later?

Reply:
I think we should commit it as-is and fix it once it is open sourced.  If we commit this as-is, I will assign an issue to myself to do this.